data-resource/benches/id_creation_benchmarks.rs

```diff
@@ -1,8 +1,21 @@
 use criterion::{black_box, criterion_group, criterion_main, Criterion};
-use data_resource::{Resource, ResourceIdTrait};
+use data_resource::ResourceId;
 use rand::prelude::*;
 use std::path::Path;
 
+#[cfg(feature = "hash_blake3")]
+use data_resource::blake3::ResourceId as Hash;
+#[cfg(not(feature = "hash_blake3"))]
+use data_resource::crc32::ResourceId as Hash;
+//todo: we have a problem with imports when we define more than 2 hash functions
+// https://users.rust-lang.org/t/conditional-module-import/82233
+//
+// somehow this causes a compilation problem because of clashing name `Hash`:
+// #[cfg(feature = "hash_blake3")]
+// use data_resource::blake3::ResourceId as Hash;
+// #[cfg(feature = "hash_crc32")]
+// use data_resource::crc32::ResourceId as Hash;
```

Comment:
I faced the same problem and couldn't find a solution for quite some time.
Just found out about [`cfg_if`](https://crates.io/crates/cfg-if) crate, which can help us manage conditional compilation features more efficiently.
It's also built by Rust engineers.